src/main/scala/org/allenai/pipeline/Producer.scala

```diff
@@ -125,7 +156,7 @@ trait CachingDisabled extends CachingEnabled {
   override def cachingEnabled: Boolean = false
 }
 
-class PersistedProducer[T, -A <: Artifact](
+class PersistedProducer[T, -A <: Artifact] private[pipeline] (
```

Comment:
Does this modifier prevent us from overriding the class in other projects?  We do use a sub-class of PersistedProducer for checkpointing in S2.